osu.Game/Overlays/BeatmapSet/Header.cs

```diff
@@ -208,9 +221,39 @@ public Header()
         }
 
         [BackgroundDependencyLoader]
-        private void load(OsuColour colours)
+        private void load(OsuColour colours, BeatmapManager beatmaps)
         {
             tabsBg.Colour = colours.Gray3;
+            this.beatmaps = beatmaps;
+
+            beatmaps.BeatmapSetAdded += handleBeatmapAdd;
+        }
+
+        protected override void Dispose(bool isDisposing)
+        {
+            base.Dispose(isDisposing);
+            beatmaps.BeatmapSetAdded -= handleBeatmapAdd;
+        }
+
+        private void handleBeatmapAdd(BeatmapSetInfo beatmap)
+        {
+            if (beatmap.OnlineBeatmapSetID == BeatmapSet.OnlineBeatmapSetID)
+                downloadButtonsContainer.FadeOut(transition_duration);

```

Comment:
shouldn't this logic also be applied if the beatmap is also in the local library? ie. the download buttons shouldn't display if i already have the beatmap at song select.